### PR TITLE
Ps 26 recipe add and recipe edit

### PR DESCRIPTION
**Pull request by Dumonoo**
Done:
* basic recipe crud
    * get by id
    * create with stages and tags
    * update
    * add Image
* basic product and tags crud + data initialization
    * create
    * get by name
    * get all
* modified many to many relationship with Association Object + added proxies to products in stage
* basic get / add / edit recipe endpoints - implementation
* add image blob in database recipe model
* added recipe View
* added recipe create form
* added recipe edit form
* setup basic recipe store with example products units

### CODE REVIEW

**Review by noFrostoo — APPROVED** [review]

**Review by lorlen — APPROVED** [review]

File: services/backend/app/api/endpoints/recipes.py [hunk 1]
```diff
@@ -88,24 +94,82 @@ def get_recipe_by_id(
...
 ) -> Any:
-    pass
+    """
+    Get recipe by id
+    """
+    recipe = crud.recipe.get_by_id(db=db, recipe_id=recipe_id)
+    if recipe:
+        return recipe
+    else:
+        raise HTTPException(status_code=400, detail="Recipe not exists.")
```
Comment (noFrostoo) [inline services/backend/app/api/endpoints/recipes.py:104]: although, maybe better return 404 here, what do you think?

**Review by noFrostoo — CHANGES_REQUESTED** [review]
welp

File: services/backend/app/api/endpoints/recipes.py [hunk 2]
```diff
@@ -88,24 +94,82 @@ def get_recipe_by_id(
...
+    recipe_id: UUID,
+    image: Optional[bytes] = File(None),
+    current_user: models.User = Depends(deps.get_current_user),
+):
+
+    recipe = crud.recipe.get_by_id(db=db, recipe_id=recipe_id)
+    if recipe:
+        crud.recipe.add_image(db=db, recipe_id=recipe_id, file=image)
+    else:
+        raise HTTPException(status_code=400, detail="Recipe not exists.")
```
Comment (noFrostoo) [inline services/backend/app/api/endpoints/recipes.py:134]: here too

File: services/backend/app/crud/crud_recipe.py [hunk 3]
```diff
@@ -66,5 +73,119 @@ def execQuery(self, q: Query, *, skip: int, limit: int) -> List[Recipe]:
...
+                    stage_id=db_stage_obj.id,
+                    amount=product.amount,
+                    amount_unit=product.amount_unit,
+                )
+                db.add(db_product_stage_obj)
+                db.commit()
+                db.refresh(db_product_stage_obj)
+
+        for tag in obj_in.tags:
+            # TODO check if tags are good
```
Comment (noFrostoo) [inline services/backend/app/crud/crud_recipe.py:109]: Tags are good, what do you mean?, also i would keep tags uniqe by name, do you agree?

File: services/backend/app/models/step.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+# from sqlalchemy.dialects.postgresql import UUID
+
+# from app.db.base_class import Base
+
+# class Step(Base):
+#     id = Column(UUID(as_uuid=True), primary_key=True, index=True, default=uuid.uuid4)
+#     content = Column(String, nullable=False)
+#     stage_id = Column(UUID(as_uuid=True), ForeignKey("stage.id"))
+
+#     stage = relationship("Stage", back_populates="steps")
```
Comment (noFrostoo) [inline services/backend/app/models/step.py:14]: remove unused code

**Review by noFrostoo — CHANGES_REQUESTED** [review]
Consider if not better to pass recpie as as prop to component from view if you have id in URL,  and not use vuex here, in other project with similar problem that was the cleaner decision, not sure not what is cleaner here

File: services/backend/app/api/endpoints/recipes.py (hunk 1 above)
Comment (Dumonoo) [reply services/backend/app/api/endpoints/recipes.py]: You are right

File: services/backend/app/api/endpoints/recipes.py (hunk 2 above)
Comment (Dumonoo) [reply services/backend/app/api/endpoints/recipes.py]: same here

File: services/backend/app/crud/crud_recipe.py (hunk 3 above)
Comment (Dumonoo) [reply services/backend/app/crud/crud_recipe.py]: It was leftover ...

File: services/backend/app/models/step.py (hunk 4 above)
Comment (Dumonoo) [reply services/backend/app/models/step.py]: unused code removed

**Review by noFrostoo — APPROVED** [review]

**Pull request merged**